File: explorer/adf.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from mysql import connector
import pandas as pd
import statsmodels.tsa.stattools as ts

class ADF:

    def __init__(self, aq):
        self.aq = aq

    def execute(self):
        self.aq.log("Start")
        mysql_connector = connector.connect(host=self.aq.DB_HOST, database=self.aq.DB_NAME,
                                            user=self.aq.DB_USR, password=self.aq.DB_PWD)
        code = pd.read_sql("""SELECT code FROM aq.future_code;""", con=mysql_connector)
        for row in code.itertuples():
            data = pd.read_sql("""SELECT datetime, open, high, low, close, volume, oi FROM future_trade
                               WHERE code='%s' AND type='%s' """ % (row[1], 5), con=mysql_connector)
            adf = ts.adfuller((data.close.diff() / data.close.shift(1))[1:], 50)
            self.aq.log("%s: ADF=%f, Lag=%d, p-value=%.10f" % (row[1], adf[0], adf[2], adf[1]))
            self.aq.log("    %s" % (adf[4]))
        mysql_connector.close()

        self.aq.log("Stop")

File: explorer/adf.r
library(RMySQL)
library(xts)
library(tseries)

rm(list=ls())

################################################################################

connect <- dbConnect(MySQL(), host="127.0.0.1", port=3306, dbname = "aq", username="root", password="!QAZ2wsx#EDC")
codes <- dbGetQuery(connect, paste0("SELECT code FROM future_code"))

################################################################################

for (code in codes$code) {
  data <- dbGetQuery(connect, paste0("SELECT datetime, open, high, low, close, volume, oi FROM future_trade 
                                      WHERE code='", code, "' AND type='5'"))
  data$datetime <- as.POSIXlt(data$datetime)
  data$rtn <- c(0, diff(data$c))/data$c
  data <- xts(data[,-1], order.by=data[,1])
  tryCatch(
    {
      print("##############################")
      adf_test <- adf.test(data$rtn)
      print(paste0(code, ": ADF=", adf_test$statistic, ", lag_order=", adf_test$parameter, ", p-value=", adf_test$p.value))
    }, error = function(e) {
      print(paste("Error ", code)) 
    })
}

################################################################################

dbDisconnect(connect)

################################################################################